File: BookieApp/BookieApp/Vistas/ViewSubirLibro/Libro.swift
//
//  Libro.swift
//  BookieApp
//
//  Created by dam2 on 20/5/24.
//

import Foundation

import Foundation

struct Libro: Codable {
    let titulo: String
    let autor: String
    let numeroPaginas: Int
    let genero: String
    let foto: String
    let sinopsis: String
    let editorial: String
    let prestado: Bool
    let filtro: [Int]
    let usuario: ModelUser
    let userId: Int
    let libroId: Int
    
    
    static func == (lhs: Libro, rhs: Libro)-> Bool{
        return lhs.libroId == rhs.libroId
    }
}


//struct ModelUser2: Identifiable, Codable {
//    var id: Int
//}

File: estechbackscaffolding-2/src/main/java/com/example/securityscaffolding/model/entity/Libro.java
package com.example.securityscaffolding.model.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;
import java.util.Set;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
@Entity
@Table(name = "libros")
public class Libro {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idLibro;

    private String titulo;

    private String autor;

    private Integer numeroPaginas;

    private String genero;

    @Column(columnDefinition = "LONGTEXT")
    private String foto;

    private String sinopsis;

    private String editorial;

    private Boolean prestado;

    private Set<Integer> filtro;

    @OneToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL, mappedBy = "libro")
    private List<ReseniaLibro> reseniaLibros = new ArrayList<>();

    @OneToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL, mappedBy = "libro")
    private List<Peticiones> peticiones = new ArrayList<>();

    @OneToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL, mappedBy = "libro")
    private List<LibrosPrestados> librosPrestados = new ArrayList<>();

    @OneToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL, mappedBy = "libro")
    private List<Favoritos> librosFavoritos = new ArrayList<>();

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_usuario", referencedColumnName = "id")
    private Usuario usuario;

}
